service/storage_service.cc

```diff
@@ -2154,11 +2154,9 @@ future<> storage_service::handle_state_normal(inet_address endpoint) {
         if (!owned_tokens.empty()) {
             on_fatal_internal_error(slogger, ::format("endpoint={} is marked for removal but still owns {} tokens", endpoint, owned_tokens.size()));
         }
+    } else if (owned_tokens.empty()) {
```

Comment:
If we had this from the start we would have not catched the bug in tests. Maybe we would catch it in prod eventually which is worse. I'd prefer the `on_internal_error` to stay so we keep catching bugs like this.

Reply:
I don't mind much keeping the internal error (I put it there in the first place),
But FWIW, in production on_internal_error would just print an error and throw, so it's still about catching the issue in testing.

Reply:
Note that here we were using `on_internal_error_noexcept` which IIUC would also crash in production.

In that case I propose we change it to `on_internal_error` as the nice middle ground - crashing the process during tests to catch bugs easier while only crashing the fiber in prod.

Reply:
No, `on_internal_error_noexcept` would not crash in production, but rather just print the error and happily continue on as if nothing happened (assuming `abort_on_internal_error` is false).
Changing to `on_internal_error` will either crash in testing or throw an error in production.